streamer/src/streamer.rs

```diff
@@ -70,14 +76,18 @@ fn recv_loop(
             if exit.load(Ordering::Relaxed) {
                 return Ok(());
             }
-            if let Ok(len) = packet::recv_from(&mut packet_batch, sock, coalesce_ms) {
+            if let Ok(len) = packet::recv_from(&mut packet_batch, sock, options.coalesce_ms) {
                 if len == NUM_RCVMMSGS {
                     num_max_received += 1;
                 }
                 recv_count += len;
                 call_count += 1;
                 if len > 0 {
-                    channel.send(packet_batch)?;
+                    if channel.len() < options.max_queued_batches {
+                        channel.send(packet_batch)?;
+                    } else {
+                        drop_count += len;
```

Comment:
I'll do that, but why did the loop `break` even when `len == 0`?

Reply:
I don't think it's actually possible for `len == 0`. In any case, reusing allocated memory makes sense here because this could be a tight loop with many iterations.